presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2334,12 +2334,17 @@ public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPl
                     .map(source::symbolToChannel)
                     .collect(toImmutableList());
 
+            Map<Integer, String> notNullChannelColumnNames = node.getColumns().stream()
+                    .filter(symbol -> node.getNotNullColumnSymbols().contains(symbol))
+                    .collect(Collectors.toMap(source::symbolToChannel, s -> node.getColumnNames().get(source.symbolToChannel(s))));
```

Comment:
You can use `toImmutableMap()` from `ImmutableMap`. Also, I think this could simplify with a `map` call:
```java
Map<Integer, String> notNullChannelColumnNames = node.getColumns().stream()
        .filter(symbol -> node.getNotNullColumnSymbols().contains(symbol))
        .map(source::symbolToChannel)
        .collect(toImmutableMap(identity(), node.getColumnNames()::get));

Reply:
The ImmutableMap is no more, but I'll remember the tip.